src/pages/settings/ExitSurvey/ExitSurveyReasonPage.tsx

```diff
@@ -0,0 +1,105 @@
+import React, {useEffect, useMemo, useState} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import FormProvider from '@components/Form/FormProvider';
+import InputWrapper from '@components/Form/InputWrapper';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import type {Choice} from '@components/RadioButtons';
+import RadioButtons from '@components/RadioButtons';
+import ScreenWrapper from '@components/ScreenWrapper';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@navigation/Navigation';
+import * as ExitSurvey from '@userActions/ExitSurvey';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type {ExitReason} from '@src/types/form/ExitSurveyReasonForm';
+import INPUT_IDS from '@src/types/form/ExitSurveyReasonForm';
+import type {Errors} from '@src/types/onyx/OnyxCommon';
+import ExitSurveyOffline from './ExitSurveyOffline';
+
+type ExitSurveyReasonPageOnyxProps = {
+    draftReason: ExitReason | null;
+};
+
+function ExitSurveyReasonPage({draftReason}: ExitSurveyReasonPageOnyxProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const {isOffline} = useNetwork();
+
+    const [reason, setReason] = useState<ExitReason | null>(draftReason);
+    useEffect(() => {
+        // disabling lint because || is fine to use as a logical operator (as opposed to being used to define a default value)
+        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+        if (reason || !draftReason) {
+            return;
+        }
+        setReason(draftReason);
+    }, [reason, draftReason]);
+    const reasons: Choice[] = useMemo(
+        () =>
+            Object.values(CONST.EXIT_SURVEY.REASONS).map((value) => ({
+                value,
+                label: translate(`exitSurvey.reasons.${value}`),
+                style: styles.mt6,
+            })),
+        [styles, translate],
+    );
```

Comment:
NAB: Why use a memo here? The reasons won't change based on responses from the API, right?

Reply:
No, the reasons won't change based on responses from the API. If it wasn't dependent on `styles` or `translate` we could move it outside the component as a static constant.

The reason I used a memo here was so that it becomes a stable reference. Without `useMemo` it would be redeclared on every render and trigger an unnecessary re-render of `InputWrapper` on every render, even though the actual reasons didn't change.

Reply:
I realize this actually isn't working anymore though since `onPress` is not stable either

Reply:
made `onPress` a stable reference too